docs/core/compatibility/core-libraries/6.0/partial-byte-reads-in-streams.md

```diff
@@ -25,38 +27,53 @@ Also, when `Stream.Read` or `Stream.ReadAsync` was called with a buffer of lengt
+
+```csharp
+var gZipBuffer = Convert.FromBase64String(compressedText);
+using var memoryStream = new MemoryStream();
+int dataLength = BitConverter.ToInt32(gZipBuffer, 0);
+memoryStream.Write(gZipBuffer, 4, gZipBuffer.Length - 4);
+var buffer = new byte[dataLength];
+memoryStream.Position = 0;
+using (var stream = new GZipStream(memoryStream, CompressionMode.Decompress))
+{
+    stream.Read(buffer, 0, buffer.Length);
+}
+```
 
 ## Version introduced
 
 6.0
 
 ## Reason for change
 
-The streams might not have returned from a read operation even if data had been successfully read. This meant they couldn't readily be used in any bidirectional communication situation where messages smaller than the buffer size were being used. This could lead to deadlocks: the application is unable to read the data from the stream that's necessary to continue the operation. It could also lead to arbitrary slowdowns, with the consumer unable to process available data while waiting for additional data to arrive.
+The streams might not have returned from a read operation even if data had been successfully read. This meant they couldn't readily be used in any bidirectional communication situation where messages smaller than the buffer size were being used. This could lead to deadlocks: the application is unable to read the data from the stream that's necessary to continue the operation. It could also lead to arbitrary slowdowns, with the consumer unable to process available data while waiting for more data to arrive.
 
 Also, in highly scalable applications, it's common to use zero-byte reads as a way of delaying buffer allocation until a buffer is needed. An application can issue a read with an empty buffer, and when that read completes, data should soon be available to consume. The application can then issue the read again, this time with a buffer to receive the data. By delegating to the wrapped stream if no already decompressed or transformed data is available, these streams now inherit any such behavior of the streams they wrap.
 
 ## Recommended action
 
 In general, code should:
 
-- Not make any assumptions about a stream `Read` or `ReadAsync` operation reading as much as was requested. The call returns the number of bytes read, which may be less than what was requested. If an application depends on the buffer being completely filled before progressing, it can perform the read in a loop to regain the behavior.
+- Not make any assumptions about a stream `Read` or `ReadAsync` operation reading as much as was requested. The call returns the number of bytes read, which might be less than what was requested. If an application depends on the buffer being completely filled before progressing, it can perform the read in a loop to regain the behavior.
 
   ```csharp
   int totalRead = 0;
   while (totalRead < buffer.Length)
   {
-      int bytesRead = stream.Read(buffer.Slice(totalRead));
+      int bytesRead = stream.Read(buffer, totalRead, buffer.Length - totalRead);
```

Comment:
I got a compile error when I tried to use Slice().

Reply:
Ah, it should be `AsSpan(totalRead)`, not `Slice(totalRead)`.

Reply:
Thanks, that worked.